### PR TITLE
Docstring update for Cartesian product and Tensor product

### LINKED ISSUE #1599: Update docstrings for `cartesian_product` and `tensor_product`.
The documentation for `cartesian_product` and `tensor_product` could use some updates. For example, see:

https://networkx.github.io/documentation/latest/reference/generated/networkx.algorithms.operators.product.cartesian_product.html#networkx.algorithms.operators.product.cartesian_product

### PR DESCRIPTION
**Pull request by OrkoHunter**
Fixes #1599.
I also used the opportunity to change the modules in `algorithms.operators` as per pep8 compilance.

### CODE REVIEW

File: networkx/algorithms/operators/binary.py [hunk 1]
```diff
@@ -58,18 +59,19 @@ def union(G, H, rename=(None, None), name=None):
     # Union is the same type as G
     R = G.__class__()
     if name is None:
-        name = "union( %s, %s )"%(G.name,H.name)
+        name = "union( %s, %s )" % (G.name, H.name)
```
Comment (SanketDG) [inline networkx/algorithms/operators/binary.py]: Use [str.format](https://docs.python.org/3/library/stdtypes.html#str.format).

Comment (OrkoHunter) [inline networkx/algorithms/operators/binary.py]: @SanketDG I think it's not a pep8 issue. Rather it's re factoring code for Python 2/3.

File: networkx/algorithms/operators/product.py [hunk 2]
```diff
@@ -181,16 +183,15 @@ def tensor_product(G, H):
...
 
-    The tensor product P of the graphs G and H has a node set that
-    is the Cartesian product of the node sets, $V(P)=V(G) \times V(H)$.
-    P has an edge ((u,v),(x,y)) if and only if (u,v) is an edge in G 
-    and x==y or  and (x,y) is an edge in H and u==v.
-    and (x,y) is an edge in H.
+    The Cartesian product P of the graphs G and H has a node set that
+    is the Cartesian product of the node sets, :math:`V(P)=V(G) \times V(H)`.
+    P has an edge ((u,v),(x,y)) if and only if either u is equal to x and
+    v & y are adjacent in H or if v is equal to y and u & x are adjacent in G.
```
Comment (chebee7i) [inline networkx/algorithms/operators/product.py]: Do you know if this definition is standard?  I've typically seen it as ((u,v), (x,y)) so long as (u,x) or (v,y) is an edge. Notably, there is no restriction that u == x or v == y.

Comment (OrkoHunter) [inline networkx/algorithms/operators/product.py]: @chebee7i I checked it on [wolfram](http://mathworld.wolfram.com/GraphCartesianProduct.html) and wrote from [the wikipedia page](https://en.wikipedia.org/wiki/Cartesian_product_of_graphs). Can you cite some sources without the restriction?

Comment (chebee7i) [inline networkx/algorithms/operators/product.py]: Opps I think I messed up. I take back my statement...seems like the operation I described would be pretty un-useful.

**Pull request merged**